src/test/ui/self/elision/ref-self-async.rs

```diff
@@ -21,13 +21,14 @@ impl<T, P> Deref for Wrap<T, P> {
 impl Struct {
     // Test using `&self` sugar:
 
-    async fn ref_self(&self, f: &u32) -> &u32 {
-        //~^ ERROR missing lifetime specifier
-        //~| ERROR cannot infer an appropriate lifetime
+    async fn ref_self(&self, f: &u32) -> &u32 { //~ ERROR lifetime mismatch
         f
     }
 
     // Test using `&Self` explicitly:
+    //
+    // FIXME(#63388). This behavior *ought* to, but does not,
```

Comment:
ick, that's pretty gnarly, though I suppose backwards-compatible (since we just disallow it, rather than misinterpreting it)

Reply:
Yeah...roughly what I was thinking. Not entirely sure this PR is the right thing to do. I'm just assuming that a more correct fix will take more time -- but I think we should definitely not just leave the code like this long term.